src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,207 @@
+                break;
+            case "email":
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void enqueueNotification(TripMonitorNotification tripMonitorNotification) {
+        notifications.add(tripMonitorNotification);
+    }
+
+    private boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
+        // Default to system's zone if one can't be inferred from trip destination.
+        ZoneId zoneId = ZoneId.systemDefault();
+        // TODO: Verify that this check doesn't cost anything.
+        Optional<ZoneId> zoneIdForCoordinates = getZoneIdForCoordinates(trip.to.lat, trip.to.lon);
+        if (zoneIdForCoordinates.isEmpty()) {
+            LOG.error("Could not find timezone for monitored trip: {}", trip.id);
+        } else {
+            zoneId = zoneIdForCoordinates.get();
+        }
+        // Get current time and trip time for comparison.
+        LocalDate now = LocalDate.now(zoneId);
+        LocalDate tripTime = trip.itinerary.startTime.toInstant()
+            .atZone(zoneId)
+            .toLocalDate();
+        // Skip check if trip is not active today or trip has already occurred.
+        if (!trip.isActiveOnDate(now) || tripTime.isAfter(now)) return true;
+        // If last check was more than an hour ago and trip doesn't occur until an hour from now, check trip.
+        long millisSinceLastCheck = System.currentTimeMillis() - trip.retrieveJourneyState().lastChecked;
+        long minutesSinceLastCheck = TimeUnit.MILLISECONDS.toMinutes(millisSinceLastCheck);
+        long minutesUntilTrip = Duration.between(tripTime.atStartOfDay(), now.atStartOfDay()).toMinutes();
+        // TODO: Refine these frequency intervals for monitor checks.
+        // If time until trip is greater than 60 minutes, we only need to check once every hour.
+        if (minutesUntilTrip > 60) {
+            // It's been about an hour since the last check. Do not skip.
+            if (minutesSinceLastCheck <= 60) {
+                return false;
+            }
+        } else {
+            // It's less than an hour until the trip time, start more frequent trip checks (about every 15 minutes).
+            if (minutesSinceLastCheck <= 15) {
+                return false;
+            }
+            // During the last 5 minutes, check every minute (assuming the loop runs every minute).
+            if (minutesUntilTrip <= 5) {
+                return false;
+            }
+        }
+        // TODO: Check that journey state is not flagged?
+        // Default to skipping check.
+        return true;
```

Comment:
Should this use the `MonitoredTrip.leadTimeInMinutes` field at all?

Reply:
Yes, will add.